connectors/sources/jira.py

```diff
@@ -320,8 +371,310 @@ def get_default_configuration(cls):
                 ],
                 "value": MAX_CONCURRENT_DOWNLOADS,
             },
+            "use_document_level_security": {
+                "display": "toggle",
+                "depends_on": [{"field": "data_source", "value": JIRA_CLOUD}],
+                "label": "Enable document level security",
+                "order": 12,
+                "tooltip": "Document level security ensures identities and permissions set in Jira are maintained in Elasticsearch. This enables you to restrict and personalize read-access users and groups have to documents in this index. Access control syncs ensure this metadata is kept up to date in your Elasticsearch documents.",
+                "type": "bool",
+                "value": False,
+            },
+        }
+
+    def _dls_enabled(self):
+        """Check if document level security is enabled. This method checks whether document level security (DLS) is enabled based on the provided configuration.
+
+        Returns:
+            bool: True if document level security is enabled, False otherwise.
+        """
+        if self._features is None:
+            return False
+
+        if not self._features.document_level_security_enabled():
+            return False
+
+        return self.configuration["use_document_level_security"]
+
+    def access_control_query(self, access_control):
+        # filter out 'None' values
+        filtered_access_control = list(
+            filter(
+                lambda access_control_entity: access_control_entity is not None,
+                access_control,
+            )
+        )
+
+        return {
+            "query": {
+                "template": {"params": {"access_control": filtered_access_control}},
+                "source": {
+                    "bool": {
+                        "filter": {
+                            "bool": {
+                                "should": [
+                                    {
+                                        "bool": {
+                                            "must_not": {
+                                                "exists": {"field": ACCESS_CONTROL}
+                                            }
+                                        }
+                                    },
+                                    {
+                                        "terms": {
+                                            f"{ACCESS_CONTROL}.enum": filtered_access_control
+                                        }
+                                    },
+                                ]
+                            }
+                        }
+                    }
+                },
+            }
+        }
+
+    def _is_active_atlassian_user(self, user_info):
+        user_url = user_info.get("self")
+        if not user_url:
+            self._logger.debug(
+                f"Skipping user: {user_info.get('displayName', 'user')} due to the absence of a personal URL."
```

Comment:
nit: As we're using `user_info.get('displayName', 'user')` at three different places I think it would be useful to have a variable `user_name`, which can be reused.

Reply:
Sure, let me wrap it in a variable.